shell/common/surface.h

```diff
@@ -15,38 +15,35 @@ namespace shell {
 
 class SurfaceFrame {
  public:
-  SurfaceFrame();
+  using SubmitCallback = std::function<bool(SkCanvas* canvas)>;
 
-  virtual ~SurfaceFrame();
+  SurfaceFrame(sk_sp<SkSurface> surface, SubmitCallback submit_callback);
+
+  ~SurfaceFrame();
 
   bool Submit();
```

Comment:
Would be nice to have a comment on this method (or perhaps on the class).  How many times should it be called?  How should the caller interpret and react to failure?